enclave/core/sgx/exception.c

```diff
@@ -385,10 +394,12 @@ void oe_virtual_exception_dispatcher(
         *arg_out = OE_EXCEPTION_CONTINUE_SEARCH;
         return;
     }
-    sgx_ssa_gpr_t* ssa_gpr =
-        (sgx_ssa_gpr_t*)(((uint8_t*)ssa_info.base_address) + ssa_info.frame_byte_size - OE_SGX_GPR_BYTE_SIZE);
+    uint64_t gprsgx_offset = (uint64_t)ssa_info.base_address +
+                             ssa_info.frame_byte_size - OE_SGX_GPR_BYTE_SIZE;
+    sgx_ssa_gpr_t* ssa_gpr = (sgx_ssa_gpr_t*)gprsgx_offset;
+    bool is_exit_info_valid = ssa_gpr->exit_info.as_fields.valid;
 
-    if (!ssa_gpr->exit_info.as_fields.valid)
+    if (!is_exit_info_valid)
```

Comment:
The following code is unsafe. We should strength the condition to: if (oe_exception_record && debug_mode)...
```c
// Consider manually provided exception information for SGX1
        if (oe_exception_record)
        {
            td->exception_address = ssa_gpr->rip;

            /* TODO PRP: We need to read the exception information from
             * oe_exception_record instead */
            td->exception_code = OE_EXCEPTION_PAGE_FAULT;
            td->exception_flags |= OE_EXCEPTION_FLAGS_HARDWARE;
        }
```
This is issue https://github.com/deislabs/mystikos/issues/399

Reply:
Thanks for the catch. I actually have another PR pending for this fix.
https://github.com/mingweishih/openenclave/commit/453da6c18a967e961c1e3d728eeb27a0e38e2953

I can add the fix to this PR as well.

Reply:
Updated, please check.